main.go

```diff
@@ -86,30 +94,128 @@ func isDockerCommand(kw string) bool {
 	return false
 }
 
-func completer(d prompt.Document) []prompt.Suggest {
-	word := d.GetWordBeforeCursor()
+//DockerHubResult : Wrap DockerHub API call
+type DockerHubResult struct {
+	PageCount        *int             `json:"num_pages,omitempty"`
+	ResultCount      *int             `json:"num_results,omitempty"`
+	ItemCountPerPage *int             `json:"page_size,omitempty"`
+	CurrentPage      *int             `json:"page,omitempty"`
+	Query            *string          `json:"query,omitempty"`
+	Items            []DockerHubImage `json:"results,omitempty"`
+}
+
+//DockerHubImage : Wrap image results of DockerHub API call
+type DockerHubImage struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	StarCount   int    `json:"star_count,omitempty"`
+	IsTrusted   *bool  `json:"is_trusted,omitempty"`
+	IsAutomated bool   `json:"is_automated,omitempty"`
+	IsOffical   *bool  `json:"is_official,omitempty"`
```

Comment:
We can replace these bool variables as non pointer type. If value is not provided we can assume it as false.
```suggestion
	IsOffical   bool  `json:"is_official"`
```

Reply:
v1 and v2 version of API is completely different. For instance, we could get the "IsOffical" item from the v1 response, rather than v2. Because of that, I've used the pointer.

Reply:
I see your point.